pagopa/io-app
# fix: [IABT-1235] Show add new payment method even when there are payable methods

## Short description
This bug occurs when the user has no favorite methods in his/her wallet.
It could happen when the user has methods but no one is set as favorite

- when user deletes a method that was set has favorite
- when user disables a method that was set has favorite

### before
https://user-images.githubusercontent.com/822471/131147850-ffd12818-d073-4970-8601-4d7fea7e453e.mov

### now
https://user-images.githubusercontent.com/822471/131147779-6264c2df-981f-4e9a-adb2-10cfecc89764.mov

This PR also fixes the margin of the debug box shown when ff `shouldDisplayVersionInfoOverlay` is enabled. This happens only on devices have notch

| before  | after |
| ------------- | ------------- |
![before](https://user-images.githubusercontent.com/822471/131147601-1b115102-1667-4410-a0cd-8037d0cb1c4c.png) | ![after](https://user-images.githubusercontent.com/822471/131147604-b96ab050-3513-4235-a2c9-f677a1eb60c8.png)

### how to test
make sure to have at least 2 methods can pay

- set at least 2 credit cards on you dev-server (you can use the web dashboard http://127.0.0.1:3000)
![Schermata 2021-08-27 alle 17 02 19](https://user-images.githubusercontent.com/822471/131148349-372ed930-2b3a-497b-9330-56295ed8612d.png)
- then press `flush`
- refresh you wallet in app
- reproduce the scenario using one of these cases:
    - delete a method that was set has favorite
    - disable a method that was set has favorite
- start a payment
